src/components/views/messages/DateSeparator.tsx

```diff
@@ -65,13 +124,127 @@ export default class DateSeparator extends React.Component<IProps> {
         }
     }
 
+    private pickDate = async (inputTimestamp): Promise<void> => {
+        const unixTimestamp = new Date(inputTimestamp).getTime();
+
+        const cli = MatrixClientPeg.get();
+        try {
+            const roomId = this.props.roomId
+            const { event_id, origin_server_ts } = await cli.timestampToEvent(
+                roomId,
+                unixTimestamp,
+                Direction.Forward
+            );
+            logger.log(`/timestamp_to_event: found ${event_id} (${origin_server_ts}) for timestamp=${unixTimestamp} (looking forward)`)
+
+            dis.dispatch({
+                action: Action.ViewRoom,
+                event_id,
+                highlighted: true,
+                room_id: roomId,
+            });
+        } catch (e) {
+            const code = e.errcode || e.statusCode;
+            // only show the dialog if failing for something other than a network error
+            // (e.g. no errcode or statusCode) as in that case the redactions end up in the
+            // detached queue and we show the room status bar to allow retry
+            if (typeof code !== "undefined") {
+                // display error message stating you couldn't delete this.
+                Modal.createTrackedDialog('Unable to find event at that date', '', ErrorDialog, {
+                    title: _t('Error'),
+                    description: _t('Unable to find event at that date. (%(code)s)', { code }),
+                });
+            }
+        }
+    };
+
+
+    private onLastWeekClicked = (): void => {
+        const date = new Date();
+        // This just goes back 7 days.
+        // FIXME: Do we want this to go back to the last Sunday? https://upokary.com/how-to-get-last-monday-or-last-friday-or-any-last-day-in-javascript/
```

Comment:
Any preference?

7 days back is simple and is probably a more expected result.

Reply:
7 days is probably fine for now. I imagine one of the first requests will be different calendar layouts though.